### ISSUE TITLE
Set developer mode after run()

### ISSUE THREAD

**Issue by avaliani** [luisfarzati/angulartics#355]
For our usage we have an app that is used both by internal users and external users. We want to avoid tracking page visits by internal users. The simplest way to make something like this happen is once we determine the user from the db to then set developer mode vs. setting it once the app is loaded. Can you consider adding this feature?

**Comment by catchshyam** [comment]
+1

**Comment by luisfarzati** [comment]
Makes sense, although I'm not sure how in the current way Angulartics works. Any ideas?

**Comment by emasch** [comment]
Currently, if the developermode is enabled, all functions are removed from the angulartics object. (@see angulartics.js#212), so it cannot be reavtivatet after doing so. There's also no watch on the property, so it doesn't recognize any changes to the property.

**Comment by emasch** [comment]
I have have a slightly different problem. As for privacy concerns, the user must be able to disable the tracking. Our management also want to give the user the ability to reactivate the tracking. So if you fix this, can you please add the functionality to reactivate the tracking? 

Maybe you make a Setting if tracking is active or not and than check it every time the directive is called. Or maybe you accept a function for that, so that we can recycle the property we allready have.

**Comment by ProLoser** [comment]
I believe we should expose a method or live cycle callback or maybe just use a $broadcast on the rootScope and you will be able to perform whatever checks you want and cancel or let you override the exposed logic so that you can monkey patch. This way we do not have to explicitly support any particular behavior

**Comment by aklimenko** [comment]
Would be great for this feature to be completed.
Btw, a question, The current code 
``if ($analytics.settings.developerMode) {
    angular.forEach($analytics, function(attr, name) {
      if (typeof attr === 'function') {
        $analytics[name] = function(){};
      }
    });
  }``

won't it remove provider method $get causing troubles?

**Comment by emasch** [comment]
For me, an event to toggle the tracking fits exactly to my needs.
